Storage/container/dictionary.h

```diff
@@ -58,31 +58,76 @@ class GenericDictionary {
     }
   }
 
+  ImmutableOptional<std::chrono::microseconds> LastModified(sfinae::CF<key_t> key) const {
+    const auto iterator = last_modified_.find(key);
+    if (iterator != last_modified_.end()) {
+      return ImmutableOptional<std::chrono::microseconds>(iterator->second);
+    } else {
+      return nullptr;
+    }
+  }
+
   void Add(const T& object) {
+    const auto now = current::time::Now();
     const auto key = sfinae::GetKey(object);
-    const auto iterator = map_.find(key);
-    if (iterator != map_.end()) {
-      const T previous_object = iterator->second;
-      journal_.LogMutation(UPDATE_EVENT(object),
-                           [this, key, previous_object]() { map_[key] = previous_object; });
+    const auto map_iterator = map_.find(key);
+    const auto lm_iterator = last_modified_.find(key);
+    if (map_iterator != map_.end()) {
+      const T& previous_object = map_iterator->second;
+      assert(lm_iterator != last_modified_.end());
+      const auto previous_timestamp = lm_iterator->second;
+      journal_.LogMutation(UPDATE_EVENT(now, object),
+                           [this, key, previous_object, previous_timestamp]() {
```

Comment:
Could you please confirm for me this `[..., previous_object, ...]` is guaranteed to capture by value, even though `previous_object` is declared as a const reference?